lib/hsa/mcwamp_hsa.cpp

```diff
@@ -2167,6 +2175,9 @@ class HSADevice final : public KalmarDevice
             throw Kalmar::runtime_exception("HCC_CHECK_COPY can only be used on machines where accelerator memory is visible to CPU (ie large-bar systems)", 0);
         }
 
+        g_agentToDeviceMap.insert(std::pair<uint64_t, HSADevice*> (agent.handle, this));
```

Comment:
This is future functionality, will be used for some upcoming debug functionality to translate the hsaAgent handle into a human-readable device#.